test/Messages.t.sol

```diff
@@ -82,26 +82,31 @@ contract MessagesTest is Test {
         );
         (uint64 decodedPoolId, bytes16 decodedTrancheId, string memory decodedTokenName, string memory decodedTokenSymbol, uint128 decodedPrice) = ConnectorMessages
             .parseAddTranche(_message.ref(0));
-        assertEq(uint256(decodedPoolId), uint256(poolId));
+        assertEq(uint(decodedPoolId), uint(poolId));
         assertEq(decodedTrancheId, trancheId);
         // Comparing raw input to output can erroneously fail when a byte string is given.
-        // Intended behaviour is that byte strings will be treated as bytes and converted to strings instead of treated as strings themselves.
-        // This conversion from string to bytes32 to string is used to simulate this intended behaviour.
+        // Intended behaviour is that byte strings will be treated as bytes and converted to strings instead
+        // of treated as strings themselves. This conversion from string to bytes32 to string is used to simulate
+        // this intended behaviour.
         assertEq(decodedTokenName, bytes32ToString(stringToBytes32(tokenName)));
         assertEq(decodedTokenSymbol, bytes32ToString(stringToBytes32(tokenSymbol)));
-        // TODO(nuno): fix this
-        assertEq(decodedPrice, uint256(price));
+        assertEq(uint(decodedPrice), uint(price));
     }
 
+    // Note: UpdateMember encodes differently in Solidity compared to the Rust counterpart because `user` is a 20-byte
+    // value in Solidity while it is 32-byte in Rust. However, UpdateMember messages coming from the cent-chain will
+    // be handled correctly as the last 12 bytes out of said 32 will be ignored.
     function testUpdateMemberEncoding() public {
         assertEq(
             ConnectorMessages.formatUpdateMember(2, bytes16(hex"811acd5b3f17c06841c7e41e9e04cb1b"), 0x1231231231231231231231231231231231231231, 1706260138),
             hex"040000000000000002811acd5b3f17c06841c7e41e9e04cb1b12312312312312312312312312312312312312310000000000000000000000000000000065b376aa"
         );
     }
 
+    // We use an UpdateMember encoded message generated in the cent-chain to
+    // verify we handle the 32 to 20 bytes address compatibility as expected.
     function testUpdateMemberDecoding() public {
-        (uint64 decodedPoolId, bytes16 decodedTrancheId, address decodedUser, uint64 decodedValidUntil) = ConnectorMessages.parseUpdateMember(fromHex("040000000000000002811acd5b3f17c06841c7e41e9e04cb1b12312312312312312312312312312312312312312312312312312312312312310000000065B376AA").ref(0));
+        (uint64 decodedPoolId, bytes16 decodedTrancheId, address decodedUser, uint64 decodedValidUntil) = ConnectorMessages.parseUpdateMember(fromHex("040000000000000002811acd5b3f17c06841c7e41e9e04cb1b12312312312312312312312312312312312312312312312312312312312312310000000065b376aa").ref(0));
```

Comment:
@offerijns note how this test uses a `UpdateMember` message generated by the cent-chain which encodes a 32-byte address. The solidity side parses the message well, which includes parsing that 32-byte address as a 20-byte one.

Ideally, both solidity and the cent-chain sides would generate the exact same `UpdateMember` message but on solidity side it will be different since the address will be encoded as a `20 bytes` + `12 zeros`.

We do have a `To Do` to make `address user` be `bytes32 user` instead and for me this is a good time to implement this. Do you agree or am I overseeing something?